api/api.go

```diff
@@ -133,113 +132,161 @@ func (api *API) ListenAndServe(addr string, tls *TLSConfig) error {
+		stats.RequestStats())
+
+	// set up middleware
+	ginjwt := middleware.JwtConfigGenerate(api.cfg.API.JwtKey, api.dbm.DB, api.l)
+	authware := []gin.HandlerFunc{
+		ginjwt.MiddlewareFunc(),
+		middleware.APIRestrictionMiddleware(api.dbm.DB),
+	}
+
+	// V1 API
+	v1 := api.r.Group("/api/v1")
+
+	// authentication
+	auth := v1.Group("/auth")
+	{
+		auth.POST("/register", api.registerUserAccount)
+		auth.POST("/login", ginjwt.LoginHandler)
+	}
+
+	// statistics
+	statistics := v1.Group("/statistics").Use(authware...)
+	{
+		statistics.GET("/stats", api.getStats)
+	}
+
+	// payments
+	payments := v1.Group("/payments", authware...)
+	{
+		payments.POST("/create", api.CreatePayment)
+		deposit := payments.Group("/deposit")
+		{
+			deposit.GET("/address/:type", api.GetDepositAddress)
+		}
+	}
+
+	// accounts
+	account := v1.Group("/account", authware...)
+	{
+		password := account.Group("/password")
+		{
+			password.POST("/change", api.changeAccountPassword)
+		}
+		key := account.Group("/key")
+		{
+			ipfs := key.Group("/ipfs")
+			{
+				ipfs.GET("/get", api.getIPFSKeyNamesForAuthUser)
+				ipfs.POST("/new", api.createIPFSKey)
+			}
+		}
+		credits := account.Group("/credits")
+		{
+			credits.GET("/available", api.getCredits)
+		}
+	}
+
+	// ipfs
+	ipfs := v1.Group("/ipfs", authware...)
+	{
+		ipfs.POST("/calculate-content-hash", api.calculateContentHashForFile)
+		ipfs.GET("/pins", api.getLocalPins)                        // admin locked
+		ipfs.GET("/check-for-pin/:hash", api.checkLocalNodeForPin) // admin locked
+		ipfs.GET("/object-stat/:key", api.getObjectStatForIpfs)
+		ipfs.POST("/download/:hash", api.downloadContentHash)
+		ipfs.POST("/pin/:hash", api.pinHashLocally)
+		addFile := ipfs.Group("/add-file")
+		{
+			addFile.POST("/", api.addFileLocally)
+			addFile.POST("/advanced", api.addFileLocallyAdvanced)
+		}
+		pubsub := ipfs.Group("/pubsub")
+		{
+			pubsub.POST("/publish/:topic", api.ipfsPubSubPublish)
+		}
+	}
+
+	// ipfs-private
+	ipfsPrivate := v1.Group("/ipfs-private", authware...)
+	{
+		ipfsPrivate.GET("/networks", api.getAuthorizedPrivateNetworks)
+		ipfsPrivate.GET("/network/:name", api.getIPFSPrivateNetworkByName) // admin locked
+		ipfsPrivate.POST("/pins", api.getLocalPinsForHostedIPFSNetwork)    // admin locked
+		ipfsPrivate.GET("/uploads/:network_name", api.getUploadsByNetworkName)
+		new := ipfsPrivate.Group("/new")
+		{
+			new.POST("/network", api.createHostedIPFSNetworkEntryInDatabase)
+		}
+		ipfsRoutes := ipfsPrivate.Group("/ipfs")
+		{
+			ipfsRoutes.POST("/check-for-pin/:hash", api.checkLocalNodeForPinForHostedIPFSNetwork) // admin locked
+			ipfsRoutes.POST("/object-stat/:key", api.getObjectStatForIpfsForHostedIPFSNetwork)
+			ipfsRoutes.POST("/pin/:hash", api.pinToHostedIPFSNetwork)
+			ipfsRoutes.POST("/add-file", api.addFileToHostedIPFSNetwork)
+			ipfsRoutes.POST("/add-file/advanced", api.addFileToHostedIPFSNetworkAdvanced)
+			ipfsRoutes.POST("/publish/details", api.publishDetailedIPNSToHostedIPFSNetwork)
```

Comment:
This should be `/api/v1/ipfs-private/ipns/publish/details`

Reply:
resolved @ 81dee0e9a769ea31dd6887d05088775fbbb473eb